app/components/floating_text_chips_input/index.tsx

```diff
@@ -0,0 +1,341 @@
+    errorIcon?: string;
+    isKeyboardInput?: boolean;
+    label: string;
+    labelTextStyle?: TextStyle;
+    onBlur?: (event: NativeSyntheticEvent<TargetedEvent>) => void;
+    onFocus?: (e: NativeSyntheticEvent<TargetedEvent>) => void;
+    onLayout?: (e: LayoutChangeEvent) => void;
+    onPress?: (e: GestureResponderEvent) => void;
+    placeholder?: string;
+    showErrorIcon?: boolean;
+    testID?: string;
+    textInputStyle?: TextStyle;
+    theme: Theme;
+    chipsValues?: string[];
+    textInputValue: string;
+    onTextInputChange: TextInputProps['onChangeText'];
+    onChipRemove: (value: string) => void;
+    onTextInputSubmitted: () => void;
+}
+
+const FloatingTextChipsInput = forwardRef<Ref, Props>(({
+    textInputValue,
+    textInputStyle,
+    onTextInputChange,
+    onTextInputSubmitted,
+    chipsValues,
+    onChipRemove,
+    theme,
+    containerStyle,
+    editable = true,
+    error,
+    errorIcon = 'alert-outline',
+    isKeyboardInput = true,
+    label = '',
+    labelTextStyle,
+    onBlur,
+    onFocus,
+    onLayout,
+    onPress,
+    placeholder,
+    showErrorIcon = true,
+    testID,
+    ...restProps
+}, ref) => {
+    const [focused, setIsFocused] = useState(false);
+    const [focusedLabel, setIsFocusLabel] = useState<boolean | undefined>();
+
+    const inputRef = useRef<TextInput>(null);
+
+    const styles = getStyleSheet(theme);
+
+    const hasValues = textInputValue.length > 0 || (chipsValues?.length ?? 0) > 0;
+
+    const shouldShowError = !focused && error;
+
+    const positions = useMemo(() => getLabelPositions(styles.textInput, styles.label, styles.smallLabel), [styles]);
+
+    // Exposes the blur, focus and isFocused methods to the parent component
+    useImperativeHandle(ref, () => ({
+        blur: () => inputRef.current?.blur(),
+        focus: () => inputRef.current?.focus(),
+        isFocused: () => inputRef.current?.isFocused() || false,
+    }), [inputRef]);
+
+    const onTextInputBlur = useCallback((e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        setIsFocusLabel(hasValues);
+        setIsFocused(false);
+
+        onBlur?.(e);
+    }, [onBlur, hasValues]);
+
+    const onTextInputFocus = useCallback((e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        setIsFocusLabel(true);
+        setIsFocused(true);
+
+        onFocus?.(e);
+    }, [onFocus]);
+
+    const onTextInputKeyPress = useCallback((e: NativeSyntheticEvent<TextInputKeyPressEventData>) => {
+        if (e.nativeEvent.key === COMMA_KEY) {
+            e.preventDefault();
+
+            inputRef.current?.clear();
+            onTextInputSubmitted();
+        }
+    }, [onTextInputSubmitted]);
+
+    function handlePressOnContainer() {
+        if (!focused) {
+            inputRef?.current?.focus();
+        }
+    }
+
+    function handleTouchableOnPress(event: GestureResponderEvent) {
+        if (!isKeyboardInput && editable && onPress) {
+            onPress(event);
+        }
+    }
```

Comment:
These are passed to props and re-created on every render. They may use a `useCallback`.